lib/ui/window/viewport_metrics.h

```diff
@@ -63,6 +67,10 @@ struct ViewportMetrics {
   double physical_view_inset_left = 0;
   double physical_view_inset_front = kUnsetDepth;
   double physical_view_inset_back = kUnsetDepth;
+  double physical_system_gesture_inset_top = 0;
```

Comment:
Are there any implications of defining gesture insets in such a central location? I'm guessing that this definition of viewport metrics is used for all platforms, right? If so, would gesture insets ever make sense on Windows, Linux, Mac, and other non-touch surfaces?

Reply:
The main reason I decided to define gesture insets this way is that I could foresee other platforms using system gesture navigations as well. I'll need to see how iOS currently handles this, but I know that swiping up from the bottom of the screen on later iPhones is analogous to pressing the home button on older models of the phone.